auth/src/main/java/org/aerogear/mobile/security/AsyncSecurityCheckExecutor.java

```diff
@@ -0,0 +1,41 @@
+package org.aerogear.mobile.security;
+
+import android.content.Context;
+
+import org.aerogear.mobile.core.metrics.MetricsService;
+import org.aerogear.mobile.security.metrics.SecurityCheckResultMetric;
+
+import java.util.Collection;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class AsyncSecurityCheckExecutor extends AbstractSecurityCheckExecutor {
+
+    private final ExecutorService pool = Executors.newFixedThreadPool(10);
+
+    AsyncSecurityCheckExecutor(final Context context, final Collection<SecurityCheck> checks, final MetricsService metricsService) {
+        super(context, checks, metricsService);
+    }
+
+    public Future<SecurityCheckResult>[] execute() {
+
+        final Collection<SecurityCheck> checks = getChecks();
+        final MetricsService metricsService = getMetricsService();
+
+        final Future[] res = new Future[checks.size()];
+
+        int i = 0;
+        for (final SecurityCheck check : checks) {
+            res[i++] = (pool.submit(() -> {
+                final SecurityCheckResult result =  check.test(getContext());
+                if (metricsService != null) {
+                    metricsService.publish(new SecurityCheckResultMetric(result));
+                }
```

Comment:
not sure i like to public a metric every time when a test is done. I prefer to send them in one go

Reply:
I think it would be a bit overkill to wait for all checks to be executed just to send the metric in one go. Do we really want that even for asynchronous execution?

Reply:
why it is overkill?